web/src/rtl.ts

```diff
@@ -139,9 +139,13 @@ export function get_direction(str) {
     return "ltr";
 }
 
-export function set_rtl_class_for_textarea($textarea) {
+export function set_rtl_class_for_textarea($textarea: JQuery<HTMLElement>): void {
     // Set the rtl class if the text has an rtl direction, remove it otherwise
     let text = $textarea.val();
+    if (typeof text !== "string") {
+        throw new TypeError(`Passed HTML element must have a
+         value attribute of type string`);
+    }
```

Comment:
Is there a way to do something like `JQuery<TextArea>` for the type of what is passed here, and have TypeScript understand that will always have a `val` that's a string? 

If we need this error, let's have the error say "HTML element must be a textarea.", since that's what the function excepts.

Reply:
@timabbott Not possible. JQuery's `.val()` method doesn't narrow down return types based on the generic type of html element, probably not implemented in its type definitions. Also, even if it were to narrow it down, it'll still only narrow it down to `string|undefined`, since it will always assume that the textarea might not exist in the DOM.

Yes, I'll change the message. Actually there's also the case of element being `undefined`, so I'll include that in the message too.

Reply:
I think it's actually better to say "must be a textarea". `undefined` is not a textarea, and I want this error to make sense to someone who might be intimidated by a fancy sounding type like `HTMLTextAreaElement`.